Firebase/Core/third_party/FIRAppEnvironmentUtil.m

```diff
@@ -152,6 +156,15 @@ + (BOOL)isFromAppStore {
 }
 
 + (BOOL)isAppStoreReceiptSandbox {
+  // Since checking the App Store's receipt URL can be memory intensive, provide an option in the
+  // Info.plist for developers to opt out of this check.
+  NSNumber *enableReceiptCheck =
+      [[NSBundle mainBundle] objectForInfoDictionaryKey:kFIRAppStoreReceiptURLEnabledKey];
+  if ([enableReceiptCheck respondsToSelector:@selector(boolValue)] &&
```

Comment:
I think we should check whether it's an NSNumber object or not. If it is an NSNumber, it should respond to boolValue. So the check should be
if (enableReceiptCheck && [enableReceiptCheck isKindOfClass:[NSNumber class]] && !enableReceiptCheck.boolValue)

Reply:
Done.